src/make-env-public.ts

```diff
@@ -0,0 +1,14 @@
+/**
+ * Make a private environment variable public, so that it can be accessed in the
+ * browser.
+ */
+export function makeEnvPublic(key: string): void {
+  // Check if the key already is already public.
+  if (/^NEXT_PUBLIC_/i.test(key)) {
+    throw new Error(`The environment variable "${key}" is already public.`);
```

Comment:
Can we instead of throwing an error do a `console.warn`? We might warn the developer that an already public environment variable is trying to be made public again, but would it be a showstopper?

Reply:
If we use this utility we're making a lot of variables public in `next.config.js`. Can we instead of calling it per variable take an array?

Instead of:
```
makeEnvPublic('SELF_URL');
makeEnvPublic('DOCKER_TAG');
makeEnvPublic('NODE_ENV');
makeEnvPublic('DEPLOYMENT_ENV');
```

Do:
```
makeEnvPublic([
  'SELF_URL',
  'DOCKER_TAG',
  'NODE_ENV',
  'DEPLOYMENT_ENV'
]);
```

Reply:
> Can we instead of throwing an error do a `console.warn`? We might warn the developer that an already public environment variable is trying to be made public again, but would it be a showstopper?

Yeah, I agree, a console warning would be more appropriate here. Worst case it would create `NEXT_PUBLIC_NEXT_PUBLIC_FOO`.

> If we use this utility we're making a lot of variables public in next.config.js. Can we instead of calling it per variable take an array?

Yes, the thought crossed my mind. Maybe allow both an array of a string?